### DISCUSSION TITLE
Dutch j-acute

### DISCUSSION THREAD

**Discussion by LettError** [koeberlin/Latin-Character-Sets#4] [General]
Thank you for this resource! 

I understand the best practice for the Dutch `/J+acute` `/j+acute` is to use the combining acute `U+0301`. This is also the method used by the iOS keyboards,.

So `j` + `U+0301` and then swap the `j` for `dotlessj` to get rid of the dot.

[This](https://github.com/koeberlin/Latin-Character-Sets#jlatnnldjlatnnld-jacutejacute) assumes that in Dutch any `j` following `iacute` is expected to be `j-acute`. This is not the always the case. 

For instance the Dutch sentence `Gaan we naar Fuji? Néé, we gaan naar Fíji!` shows a context in which the `j` should not become `j-acute`.

**Comment by koeberlin** [comment]
Many thanks for your feedback, Erik!

So would you say a combination of `ccmp` (for getting rid of the dot) and `mark` (to place the acute) might be desirable, while the Durch `locl` code could be dropped?

**Comment by colinmford** [comment]
My approach to supporting the jacute is as follows, your mileage my vary. This at least fulfills Apple's requirements for "support" of Dutch. When the J/jacute is typed on a Dutch iOS keyboard (by holding on `j` and selecting it in the pop-up), it places `j` + `U+0301` as Erik mentioned. Fonts without `U+0301` now do not technically support Dutch according to Apple, and may not show up in font menus if the user's system language is set to Dutch.

1. Include `J`, `j`, `dotlessj` (U+0237, or no unicode, this glyph is purely to remove the dot and is not needed for Dutch support), and `acutecmb` (U+0301)
2. `ccmp` as follows:
```
feature ccmp {
    ...
    sub j' acutecmb by dotlessj;
    ...
} ccmp;
```
3. set up anchors for `J`, `dotlessj`, and `acutecmb` so that a `mark` feature can be generated for placement.

I don't use a `languagesystem latn NLD` set up or anything like that.

**Reply by koeberlin** [reply]
Thank you, Colin – that’s quite what I have in mind too. (Just been waiting for Erik’s reply)
I tend to suggest a combination of both though, to maximize support.

**Reply by moyogo** [reply]
The `sub j' acutecomb by dotlessj;` should be generalized for any top combining mark that replaces the dot. Some rare Dutch texts used ij with two grave accents for example, see https://commons.m.wikimedia.org/wiki/Category:J%CC%80 , other systems also use j with other top accents. This should also be generalized for other soft dotted letters like į, ị, ɨ when included as they should lose their dot when combined with to accents as well.

**Reply by koeberlin** [reply]
> This should also be generalized for other soft dotted letters like į, ị, ɨ when included as they should lose their dot when combined with to accents as well.

Agreed. Glyphs does this automatically.

**Reply by colinmford** [reply]
@moyogo Yes, I typically also write the ccmp feature to accommodate other dotted letters, the code was just a sample for this particular situation.

**Reply by moyogo** [reply]
@colinmford Got it, my bad.
@koeberlin Unfortunately Glyphs only does this for i, j, but not for į, ị, ḭ, ɨ, і (i-cy), for example į́ in Navajo, ị́ in Igbo when tones are marked, ḭ́ in some African languages, ɨ́ in some American or African languages, і́ in Ukrainian when stress is marked, or other phonetic or theoretical ones like ᶖ, ɉ, ј (je-cy), ⁱ, etc. that all have the Unicode Soft_Dotted property.

**Reply by LettError** [reply]
Please do these things for living languages! Joost van den Vondel is quite dead ;)

**Reply by weiweihuanghuang** [reply]
> @colinmford Got it, my bad. @koeberlin Unfortunately Glyphs only does this for i, j, but not for į, ị, ḭ, ɨ, і (i-cy), for example į́ in Navajo, ị́ in Igbo when tones are marked, ḭ́ in some African languages, ɨ́ in some American or African languages, і́ in Ukrainian when stress is marked, or other phonetic or theoretical ones like ᶖ, ɉ, ј (je-cy), ⁱ, etc. that all have the Unicode Soft_Dotted property.

Just to clarify does that mean something like į iogonek needs to be swapped for /dotlessi/ogonekcomb to make ı̨́ like:
`sub iogonek' acutecomb by dotlessi ogonekcomb acutecomb;`

**Reply by colinmford** [reply]
@weiweihuanghuang or if the ogonek has to be attached smoothly for the design if the `iogonek` (as some times has to occur) you would make a `dotlessiogonek` (or however you want to name that) glyph, then the feature would be something like
```
@soft_dotted = [i        j        iogonek];
@dotless     = [dotlessi dotlessj dotlessiogonek];

sub @soft_dotted' acutecmb by @dotless;
```
And expand those classes to include any other soft-dotted glyphs you might need, as @moyogo alludes to.
